File: Lab1.py
def happy(ticket_number):
    first_half = int(str(ticket_number)[: len(str(ticket_number)) // 2])
    second_half = int(str(ticket_number)[len(str(ticket_number)) // 2:])
    return sum(map(int, str(first_half))) == sum(map(int, str(second_half)))


flag = True
while flag:
    inp_dat  = int(input("Enter the number of digits: "))
    #*while True:
    #    try:
    #        inp_dat  = int(input("Enter the number of digits: "))
    #    except ValueError:
    #        print("Incorrect input")
    #        continue
    #   if inp_dat // 2 !=0:
    #        break
    #    
    count = 0
    for ticket_number in range(10 ** (inp_dat // 2), 10 ** inp_dat):
        if happy(ticket_number):
            count += 1
    print("quantity of happy ticket = ", count)
    d = int(input("Continue?: 0. Нет  1. Да\n"))
    if d == 0:
        flag = False
print("Exit")

File: Lab1.cpp
#include<iostream>
#include <cmath>
using namespace std;


int main() { 
    
    int inp_data, sum1, sum2;
    bool select = 1;
	while(select) {
        cout << "Enter even number length\n";
	    cin >> inp_data;
	    
        while (cin.fail() || inp_data % 2 !=0 ||inp_data <= 0) // проверка длины
	    {
		    cin.clear();
		    cin.ignore(1000, '\n');
		    cout << "incorrect input\n" << "Enter even number length\n"; cin >> inp_data;
	    }
	    
        int count = 0;
        for(int first_half = 0; first_half < pow(10, inp_data / 2); first_half++){ //цикл для чисел первой половины
            for(int second_half = 0;second_half < pow(10, inp_data / 2); second_half++){ //цикл для чисел второй половины
                sum1 = 0;
                sum2 = 0;
                int tmp_1st = first_half;
                int tmp_2nd = second_half;
                for(int digit = 0; digit < inp_data / 2; ++digit) { //цикл для разбиения на числа
                    sum1 += tmp_1st % 10;
                    sum2 += tmp_2nd % 10;
                    tmp_1st /= 10;
                    tmp_2nd /= 10;
                }
                if (sum1 == sum2) { // если сумма первой половины равна сумме второй, то прибавляем счётчик
                    count++;
                }
            }
        }
        
        cout << "count of happy ticket = "<< count << endl 
        << "continue?: 0.No 1.Yes\n";
        cin >> select;
        while (cin.fail()) { // проверка ввода
            cin.clear(); 
            cin.ignore(10000, '\n'); 
            cout << "incorect input. try again \n"; 
            cin >> select; 
        }
	}
	cout << "Exit";
    return 0;
} 